### DISCUSSION TITLE
Shortcut to undo/delete a recently expanded snippet in LuaSnip

### DISCUSSION THREAD

**Discussion by kylefhartzenberg** [L3MON4D3/LuaSnip#973] [Q&A] [answered]
Say a user invokes a snippet, but realises that the snippet is incorrect/undesired. Is there the ability to clear the most recently expanded snippet via the LuaSnip API such that this can then be bound to a key-mapping in Neovim? For example (psuedo-code):

```lua
vim.keymap.set({'i', 's'}, 'uu', '<cmd>lua require("luasnip").undo_expansion()<CR>')
```

The documentation doesn't seem to mention anything relating to undoing or deleting a recently expanded snippet. The closest discussion I could find in the repository was https://github.com/L3MON4D3/LuaSnip/issues/116#issuecomment-903704897 which mentions a function that "*reverses the expansion of a snippet and puts the trigger-text back, mapping that to a key seems pretty ergonomic*". This is basically what I am after (preferably with the option to *not* put the trigger-text back as well, as in some cases, one wants to remove the snippet (and trigger) because they have changed their mind about what needs to be typed/inserted).

**Comment by L3MON4D3** [comment, marked as answer]
#797!
I haven't gotten to implementing it yet, but I'll do so, it seems desired :D

**Reply by kylefhartzenberg** [reply]
Thank you. I actually forgot that I went down the rabbit hole of https://github.com/L3MON4D3/LuaSnip/issues/797, https://github.com/L3MON4D3/LuaSnip/issues/830 and https://github.com/L3MON4D3/LuaSnip/issues/959 and had set-up `<C-o>u` following https://github.com/L3MON4D3/LuaSnip/issues/830#issuecomment-1489967687, but, as mentioned therein, the undo point is set at the beginning of a snippet-expansion-bonanza which means that one could be typing some TeX commands (as an example) and be 5+ snippets in and it'll undo it all. If I get some time, I'll play around with trying to implement the steps you listed in https://github.com/L3MON4D3/LuaSnip/issues/797#issuecomment-1447761739. Appreciate the help and your work!
